### DISCUSSION TITLE
sea-orm-cli truncates mysql passwords with special characters

### DISCUSSION THREAD

**Discussion by Ajkcki** [SeaQL/sea-orm#1263] [Q&A] [answered]
### sea-orm-cli 0.9.3 couldn't connect to local mysql server.

In .env file, 
DATABASE_URL="mysql://user1:vnsiwQci23$s@localhost/db1"

debug to `sea-orm-cli 0.9.3`/src/cli.rs , and the url string became

"mysql://user1:vnsiwQci23@localhost/db1"

Notice the password was truncated at char `$`.

It is in this function in src/cli.rs,  the truncated url was found. 
```
pub async fn run_cli<M>(migrator: M)
where
    M: MigratorTrait,
{
    dotenv().ok();
    let url = std::env::var("DATABASE_URL").expect("Environment variable 'DATABASE_URL' not set");
    let db = &Database::connect(&url).await.unwrap();
    let cli = Cli::parse();

    run_migrate(migrator, db, cli.command, cli.verbose)
        .await
        .unwrap_or_else(handle_error);
}
```

**Comment by billy1624** [comment]
Hey @Ajkcki, try to encode the `$` as `%24`. That means your connection string would be... `DATABASE_URL="mysql://user1:vnsiwQci23%24s@localhost/db1"`.

Btw... I think one should avoid using `" ' $ , [ ] * ? { } ~ # % \ < > | ^ ;` as password of database user to avoid this pitfall.

**Comment by Ajkcki** [comment]
Mysql `Validate Password Plugin` specifically requires passwords contain at least on special char, it will be very difficult to avoid all these chars. In addition,  the same url string with char `$` works in `sqlx`,  upon which sea-orm is built.  It would be reasonable to expect sea-orm to accept such a string as well.

**Comment by billy1624** [comment, marked as answer]
Okay, changing the double quote `DATABASE_URL="mysql://user1:vnsiwQci23$s@localhost/db1"` into single quote should solve the issue `DATABASE_URL='mysql://user1:vnsiwQci23$s@localhost/db1'`.

The dollar sign is being treated as a variant in bash, i.e. `$s`. The variable will be evaluated when wrapped inside double quote but not when wrapped inside single quote.
